Allow underscores in names

holograph opened the pull request:
As the name implies, this adds support for underscore in various names. I'm not entirely sure why this model is so strict to begin with (the [Go SDK](https://github.com/helm/helm/blob/main/pkg/chart/v2/metadata.go#L93) only strips non-printable characters and leading/tailing whitespace), but I can attest that underscores work perfectly fine in e.g. chart dependency names, which is where I ran into this :-)

holograph committed:
Allow underscores in names

JohnGarbutt commented on pyhelm3/models.py:
```diff
@@ -43,7 +43,7 @@ def __init__(self, _command: Command, **kwargs):
 
 #: Type for a name (chart or release)
-Name = constr(pattern = r"^[a-z0-9-]+$")
+Name = constr(pattern = r"^[a-z0-9_-]+$")
```
I think its because of the Chart name restrictions here:
https://helm.sh/docs/chart_best_practices/conventions/#chart-names

Maybe we need a different type for the release names, which are not as strict, it would seem?

tomerg-qm replied:
I've actually looked into our use case a bit more carefully, apparently (per the [documentation](https://helm.sh/docs/topics/charts/#alias-field-in-dependencies)) installing a chart that has an alias dependency will, and I quote here, "put a chart in dependencies using alias as name of new dependency."

Our issue stems from getting an installed chart's metadata from a live cluster. We use [mimir-distributed 5.3.0](https://github.com/grafana/mimir/blob/mimir-distributed-5.3.0/operations/helm/charts/mimir-distributed/Chart.yaml#L20-L24) which installs `rollout_operator`. Calling `pyhelm3.Client(...).get_current_revision(...).chart_metadata()` blows up because the returned dependency name is actually the declared alias `rollout_operator`, which includes an underscore.

I've pushed another commit to better express this, please have a look @JohnGarbutt

benbz replied:
I've run into a similar problem with a dependency that has an [`alias` with upper-case characters](https://github.com/goauthentik/helm/blob/main/charts/authentik/Chart.yaml#L40-L44) which is blocked by the same restriction

JohnGarbutt reviewed (changes requested):
I think we might need to split chart names and release names, each having their own type, to make sure we are keep correctly checking the chart names.

Would that work OK for your use case?

holograph committed:
Enable dependency names to include underscores

Aliases may sometimes include underscores, and in turn may be returned a dependency name when using helm.get_current_revision(...).chart_metadata().
